lib/appSettings.ts

```diff
@@ -0,0 +1,162 @@
+import { prisma, prismaErrors } from "@lib/integration/prismaConnector";
+import { AccessControlError, checkPrivileges } from "./privileges";
+import { logEvent } from "./auditLogs";
+import { logMessage } from "@lib/logger";
+import { UserAbility } from "./types";
+import { settingCheck, settingPut, settingDelete } from "@lib/cache/settingCache";
+
+export const getAllAppSettings = async (ability: UserAbility) => {
+  try {
+    checkPrivileges(ability, [{ action: "view", subject: "Setting" }]);
+    const settings = await prisma.setting
+      .findMany({
+        select: {
+          internalId: true,
+          nameEn: true,
+          nameFr: true,
+          descriptionEn: true,
+          descriptionFr: true,
+          value: true,
+        },
+      })
+      .catch((e) => prismaErrors(e, []));
+    logEvent(ability.userID, { type: "Setting" }, "ListAllSettings");
```

Comment:
I thought about it as well but the only time that function would be called is when you visit the admin settings page.  Given the low frequency, I was on the fence about caching those values.  Open to suggestions though if we're worried it might cause undue stress on the DB.